src/modules/organisation-role/organisation-role.service.ts

```diff
@@ -126,6 +132,55 @@ export class OrganisationRoleService {
     }
   }
 
+  async removeRole(organisationId: string, roleId: string) {
+    const role = await this.rolesRepository.findOne({
+      where: { id: roleId, organisation: { id: organisationId }, isDeleted: false },
+    });
+    if (!role) {
+      throw new NotFoundException(`The role with ID ${roleId} does not exist`);
+    }
+    const usersWithRole = await this.organisationRepository.count({
+      where: {
+        id: organisationId,
+        organisationMembers: {
+          role: { id: roleId },
+        },
+      },
+      relations: ['organisationUsers', 'organisationUsers.role'],
+    });
+
+    if (usersWithRole > 0) {
+      throw new BadRequestException('Role is currently assigned to users');
+    }
+    try {
```

Comment:
There is a response interceptor for this. You can remove everything within the catch block, and make your dB call outside the try block.